gef.py

```diff
@@ -10597,10 +10597,34 @@ def read_cstring(self,
         encoding = encoding or "unicode-escape"
         length = min(address | (DEFAULT_PAGE_SIZE-1), max_length+1)
 
+        while not self.get_section(address + length - 1) and length > 0:
```

Comment:
I didn't measure it but I think this approach may cause a slow down in `read_cstring` for long printable strings. I also find quite complex for what is actually intended to do because you'll be impacting all cases of use of this function for only a minority of cases that violate page boundary.

A simpler approach would be to remove all this, simply let `self.read` execute: if a gdb.MemoryError, use the exception information calculate the maximum valid length and re-read.

Another approach would be to check early on when setting `length` that address + length is a readable address.